action_viafence/viafence_action.py

```diff
@@ -124,79 +134,90 @@ def mainDialogToSelf(self):
-        self.isSameNetZoneViasOnlyChecked = 0
-
-        self.mainDlg = MainDialog(None)
-        self.selfToMainDialog()
-
-        if (self.mainDlg.ShowModal() == wx.ID_OK):
-            # User pressed OK.
-            # Assemble a list of pcbnew.BOARD_ITEMs derived objects that support GetStart/GetEnd and IsOnLayer
-            self.mainDialogToSelf()
-            lineObjects = []
-
-            # Do we want to include net tracks?
-            if (self.isNetFilterChecked):
-                # Find nets that match the generated regular expression and add their tracks to the list
-                netRegex = self.regExFromSimpleEx(self.netFilter)
-                for netId in self.netMap:
-                    if re.match(netRegex, self.netMap[netId].GetNetname()):
-                        for trackObject in self.boardObj.TracksInNet(netId):
-                            lineObjects += [trackObject]
-
-            # Do we want to include drawing segments?
-            if (self.isIncludeDrawingChecked):
-                boardItem = self.boardObj.GetDrawings().GetFirst()
-                while boardItem is not None:
-                    if pcbnew.DRAWSEGMENT.ClassOf(boardItem):
-                        # A drawing segment (not a text or something else)
-                        drawingObject = boardItem.Cast()
-                        if drawingObject.GetShape() == pcbnew.S_SEGMENT:
-                            # A straight line
-                            lineObjects += [drawingObject]
-
-                    boardItem = boardItem.Next()
-
-            # Do we want to filter the generated lines by layer?
-            if (self.isLayerChecked):
-                # Filter by layer
-                # TODO: Make layer selection also a regex
-                lineObjects = [lineObject for lineObject in lineObjects if lineObject.IsOnLayer(self.layerId)]
-
-            # Generate a path list from the pcbnew.BOARD_ITEM objects
-            self.pathList =  [[ [lineObject.GetStart()[0], lineObject.GetStart()[1]],
-                                [lineObject.GetEnd()[0],   lineObject.GetEnd()[1]]   ]
-                                for lineObject in lineObjects]
-
-            # Generate via fence
-            try:
-                viaPoints = generateViaFence(self.pathList, self.viaOffset, self.viaPitch)
-            except:
-                viaPoints = []
-
-            if (self.isDebugDumpChecked):
-                self.dumpJSON(os.path.join(self.boardPath, time.strftime("viafence-%Y%m%d-%H%M%S.json")))
-
-            viaObjList = self.createVias(viaPoints, self.viaDrill, self.viaSize, self.viaNetId)
+        #check for pyclipper lib
+        pyclip = False
+        try:
+            import pyclipper
+            pyclip = True
+        except:
+            wdlg = wx.MessageDialog(None, u"\u2718 ERROR Missing KiCAD \'pyclipper\' python module",'ERROR message',wx.OK | wx.ICON_ERROR)
+            result = wdlg.ShowModal()
+        if pyclip:
```

Comment:
Why use the pyclip variable for this? This looks too complicated for what it does.

Reply:
done with this commit
https://github.com/skuep/kicad-plugins/pull/8/commits/0ec66aee348139336e70d31d271eb7cc644a8695